python/lsst/ip/isr/deferredCharge.py

```diff
@@ -210,6 +220,479 @@ def capture(self, pixel_signals):
+    traps : `list` [`lsst.ip.isr.SerialTrap`]
+        Serial traps to simulate.
+    """
+
+    def __init__(self, imarr, prescan_width, output_amplifier, cti=0.0, traps=None):
+        # Image array geometry
+        self.prescan_width = prescan_width
+        self.ny, self.nx = imarr.shape
+
+        self.segarr = np.zeros((self.ny, self.nx+prescan_width))
+        self.segarr[:, prescan_width:] = imarr
+
+        # Serial readout information
+        self.output_amplifier = output_amplifier
+        if isinstance(cti, np.ndarray):
+            raise ValueError("cti must be single value, not an array.")
+        self.cti = cti
+
+        self.serial_traps = None
+        self.do_trapping = False
+        if traps is not None:
+            if not isinstance(traps, list):
+                traps = [traps]
+            for trap in traps:
+                self.add_trap(trap)
+
+    def add_trap(self, serial_trap):
+        """Add a trap to the serial register.
+        Parameters
+        ----------
+        serial_trap : `lsst.ip.isr.SerialTrap`
+            The trap to add.
+        """
+        try:
+            self.serial_traps.append(serial_trap)
+        except AttributeError:
+            self.serial_traps = [serial_trap]
+            self.do_trapping = True
+
+    def ramp_exp(self, signal_list):
+        """Simulate an image with varying flux illumination per row.
+        This method simulates a segment image where the signal level
+        increases along the horizontal direction, according to the
+        provided list of signal levels.
+        Parameters
+        ----------
+        signal_list : `list` [`float`]
+            List of signal levels.
+        Raises
+        ------
+        ValueError
+            Raised if the length of the signal list does not equal the
+            number of rows.
+        """
+        if len(signal_list) != self.ny:
+            raise ValueError("Signal list does not match row count.")
+
+        ramp = np.tile(signal_list, (self.nx, 1)).T
+        self.segarr[:, self.prescan_width:] += ramp
+
+    def readout(self, serial_overscan_width=10, parallel_overscan_width=0):
+        """Simulate serial readout of the segment image.
+        This method performs the serial readout of a segment image
+        given the appropriate SerialRegister object and the properties
+        of the ReadoutAmplifier.  Additional arguments can be provided
+        to account for the number of desired overscan transfers. The
+        result is a simulated final segment image, in ADU.
+        Parameters
+        ----------
+        serial_overscan_width : `int`, optional
+            Number of serial overscan columns.
+        parallel_overscan_width : `int`, optional
+            Number of parallel overscan rows.
+        Returns
+        -------
+        result : `np.ndarray` (nx, ny)
+            Simulated image, including serial prescan, serial
+            overscan, and parallel overscan regions.
```

Comment:
Can you add a note on this docstring about what units this expects for input and output?  Also there should be an empty line before `Parameters`.